### PR TITLE
GH-2510 bump version of cache action and use in status CI jobs as well

### PR DESCRIPTION
**Pull request by jeenbroekstra**
GitHub issue resolved: #2510 

Briefly describe the changes proposed in this PR:

- bump to new version of cache action
- add config to all build jobs, not just pr verify

This may need some tweaking / testing. 

---- 
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/master/.github/CONTRIBUTING.md) for more details):

 - [ ] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [ ] I've added tests for the changes I made
 - [ ] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/master/.github/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [ ] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change
 - [ ] every commit has been [signed off](https://stackoverflow.com/questions/1962094/what-is-the-sign-off-feature-in-git-for)

Note: we merge all feature pull requests using [squash and merge](https://help.github.com/en/github/administering-a-repository/about-merge-methods-on-github#squashing-your-merge-commits). See [RDF4J git merge strategy](https://rdf4j.org/documentation/developer/merge-strategy/) for more details.

### CODE REVIEW

**Comment by jeenbroekstra** [comment]
Looks like the setup works. But then I don't quite see why the older setup didn't also work. Unless it was to do with the fact that it was only use on the pr verify jobs.

File: .github/workflows/develop-status.yml [hunk 1]
```diff
@@ -19,6 +19,13 @@ jobs:
       uses: actions/setup-java@v1
       with:
         java-version: ${{ matrix.jdk }}
+    - name: Cache local Maven repository
+      uses: actions/cache@v2
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
```
Comment (hmottestad) [inline .github/workflows/develop-status.yml:28]: What is the difference between key and restore key? Are they supposed to be different?

**Comment by hmottestad** [comment]
Screenshot 2020-09-17 at 07 34 49

Does this mean that it never overwrites the cache? Or does it overwrite the cache?

Do you know if the caching step is skipped if the build fails?

**Comment by jeenbroekstra** [comment]
> Does this mean that it never overwrites the cache? Or does it overwrite the cache?

I think that error happens because we have two parallel builds on the same PR, who are therefore using the same cache entry key. This error happens on the java 14 build because the other build (1.8) already pushed a key. Perhaps we can adjust the key to make it specific to each build. 

> Do you know if the caching step is skipped if the build fails?

No, I don't know.

File: .github/workflows/develop-status.yml (hunk 1 above)
Comment (jeenbroekstra) [reply .github/workflows/develop-status.yml:28]: See https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows . I just copy-pasted this setup. As I inderstand it the restore keys are optional, and can be used as alternative lookup keys if the main key gives no hit.

**Comment by jeenbroekstra** [comment]
@hmottestad feel free to play with this a little bit as well btw, I am not able to spend more time on this today or tomorrow I'm afraid.

**Comment by hmottestad** [comment]
Will do. Thanks for putting this up here :)

**Comment by jeenbroekstra** [comment]
@hmottestad I think I have an idea for making that error you saw above go away. Going to push a small fix.

**Comment by jeenbroekstra** [comment]
Ok so that answers that question: if the build fails, the post cache update step is skipped.

**Review by jeenbroekstra — COMMENTED** [review]
I'm happy with this now - looks like the adapted cache key make the system more robust.

**Comment by hmottestad** [comment]
I triggered a re-run of the ci and I can see that it's not downloading the cached files. So as long as we do not update the pom.xml files it should use the cache.

**Review by hmottestad — APPROVED** [review]

**Comment by jeenbroekstra** [comment]
You sure? Looks to me as if it does download the cache.

**Comment by hmottestad** [comment]
Hahahaha. I meant that it does download the cache. Not sure where the "not" came from :P Sorry for the confusion.

**Pull request merged**